codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -748,13 +748,18 @@ public boolean canOpenStream() {
         public DefaultStream createStream(int streamId, boolean halfClosed) throws Http2Exception {
             State state = activeState(streamId, IDLE, isLocal(), halfClosed);
 
-            checkNewStreamAllowed(streamId, state);
+            try {
+                checkNewStreamAllowed(streamId, state);
+            } catch (Http2Exception e) {
+                throw e;
+            } finally {
```

Comment:
There is no need to rethrow. 
```suggestion
            } finally {
```

Reply:
Oh cool did not know that was the behavior but good to know - thanks @normanmaurer 

Updated ⚡